index.js

```diff
@@ -173,4 +173,57 @@ mainForm.addEventListener('submit', (e) => {
     return false;
   }
   return true;
-});
\ No newline at end of file
+});
+
+// fetching enterd details from user
+const providedEmail = document.getElementById('email-input');
+const messageFromUser = document.getElementById('text-input');
+const dataByUser = {};
+const providedName = document.getElementById('name-input');
+
+// Local Data storage
+function isStorageSupported(type) {
+  let localStorage;
+  try {
+    localStorage = window[type];
+    const i = '__testing_storage__';
+    localStorage.setItem(i, i);
+    localStorage.removeItem(i);
+    return true;
+  } catch (e) {
+    return (
+      e instanceof DOMException
+      && (e.code === 22
+        || e.code === 1014
+        || e.name === 'QuotaExceededError'
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+        && localStorage
+        && localStorage.length !== 0);
+  }
+}
+
+function siteUpdater() {
+  const currentUserData = JSON.parse(localStorage.getItem('dataByUser'));
+  providedName.value = currentUserData.name ? currentUserData.name : '';
+  providedEmail.value = currentUserData.email ? currentUserData.email : '';
+  messageFromUser.value = currentUserData.textarea ? currentUserData.textarea : '';
+}
```

Comment:
- [ ] Kindly ensure that when visiting the page the form inputs should be auto-populated with the values entered in the previous visit to the page and If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty 👍🏽 .

Reply:
Thank you